errors when installing from source

some onnxruntime errors happend
```
[ 72%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-microphone-offline.dir/microphone.cc.o
../../_deps/onnxruntime-src/lib/libonnxruntime_providers_cuda.so: undefined reference to `Provider_GetHost'
collect2: error: ld returned 1 exit status
make[2]: *** [bin/sherpa-onnx-offline] Error 1
make[1]: *** [sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-offline.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 73%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-microphone.dir/microphone.cc.o
[ 74%] Linking CXX executable ../../bin/sherpa-onnx-microphone-offline
../../_deps/onnxruntime-src/lib/libonnxruntime_providers_cuda.so: undefined reference to `Provider_GetHost'
collect2: error: ld returned 1 exit status
make[2]: *** [bin/sherpa-onnx-microphone-offline] Error 1
make[1]: *** [sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-microphone-offline.dir/all] Error 2
[ 75%] Linking CXX executable ../../bin/sherpa-onnx-microphone
../../_deps/onnxruntime-src/lib/libonnxruntime_providers_cuda.so: undefined reference to `Provider_GetHost'
collect2: error: ld returned 1 exit status
make[2]: *** [bin/sherpa-onnx-microphone] Error 1
make[1]: *** [sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-microphone.dir/all] Error 2
[ 76%] Linking CXX executable ../../bin/sherpa-onnx
../../_deps/onnxruntime-src/lib/libonnxruntime_providers_cuda.so: undefined reference to `Provider_GetHost'
collect2: error: ld returned 1 exit status
make[2]: *** [bin/sherpa-onnx] Error 1
make[1]: *** [sherpa-onnx/csrc/CMakeFiles/sherpa-onnx.dir/all] Error 2
make: *** [all] Error 2
```

I followed
```
git clone https://github.com/k2-fsa/sherpa-onnx
cd sherpa-onnx
mkdir build
cd build
cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON -DSHERPA_ONNX_ENABLE_GPU=ON ..
make -j6
```
the output of cmake is:
```
cmake -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=ON -DSHERPA_ONNX_ENABLE_GPU=ON ..
-- The C compiler identification is GNU 8.3.1
-- The CXX compiler identification is GNU 8.3.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/rh/devtoolset-8/root/bin/gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/rh/devtoolset-8/root/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Warning at CMakeLists.txt:62 (message):
  Compiling for NVIDIA GPU is enabled.  Please make sure cudatoolkit

  is installed on your system.  Otherwise, you will get errors at runtime.

  Hint: You don't need sudo permission to install CUDA toolkit.  Please refer
  to

    https://k2-fsa.github.io/k2/installation/cuda-cudnn.html

  to install CUDA toolkit if you have not installed it.


-- CMAKE_BUILD_TYPE: Release
-- CMAKE_INSTALL_PREFIX: /usr/local
-- BUILD_SHARED_LIBS ON
-- SHERPA_ONNX_ENABLE_PYTHON OFF
-- SHERPA_ONNX_ENABLE_TESTS OFF
-- SHERPA_ONNX_ENABLE_CHECK OFF
-- SHERPA_ONNX_ENABLE_PORTAUDIO ON
-- SHERPA_ONNX_ENABLE_JNI OFF
-- SHERPA_ONNX_ENABLE_C_API ON
-- SHERPA_ONNX_ENABLE_WEBSOCKET ON
-- SHERPA_ONNX_ENABLE_GPU ON
-- Looking for C++ include alsa/asoundlib.h
-- Looking for C++ include alsa/asoundlib.h - not found
-- Looking for C++ include cxxabi.h
-- Looking for C++ include cxxabi.h - found
-- Looking for C++ include execinfo.h
-- Looking for C++ include execinfo.h - found
-- Found local downloaded kaldi-native-fbank: /home/sunty/workspace/k2_env/sherpa-onnx/kaldi-native-fbank-1.18.1.tar.gz
-- Downloading kaldi-native-fbank from /home/sunty/workspace/k2_env/sherpa-onnx/kaldi-native-fbank-1.18.1.tar.gz
-- kaldi-native-fbank is downloaded to /home/sunty/workspace/k2_env/sherpa-onnx/build/_deps/kaldi_native_fbank-src
-- kaldi-native-fbank's binary dir is /home/sunty/workspace/k2_env/sherpa-onnx/build/_deps/kaldi_native_fbank-build
-- CMAKE_BUILD_TYPE: Release
-- CMAKE_EXPORT_COMPILE_COMMANDS: 
-- BUILD_SHARED_LIBS: ON
-- KALDI_NATIVE_FBANK_BUILD_TESTS: OFF
-- KALDI_NATIVE_FBANK_BUILD_PYTHON: OFF
-- KALDI_NATIVE_FBANK_ENABLE_CHECK: OFF
-- KALDI_NATIVE_FBANK_ENABLE_CHECK: OFF
-- CMAKE_CXX_FLAGS: 
-- CMAKE_INSTALL_PREFIX: /usr/local
-- Looking for C++ include cxxabi.h
-- Looking for C++ include cxxabi.h - found
-- Looking for C++ include execinfo.h
-- Looking for C++ include execinfo.h - found
-- Disable building Python
-- CMAKE_SYSTEM_NAME: Linux
-- CMAKE_SYSTEM_PROCESSOR: x86_64
-- location_onnxruntime_header_dir: location_onnxruntime_header_dir-NOTFOUND
-- location_onnxruntime_lib: location_onnxruntime_lib-NOTFOUND
-- location_onnxruntime_cuda_lib: location_onnxruntime_cuda_lib-NOTFOUND
-- Could not find a pre-installed onnxruntime. Downloading pre-compiled onnxruntime
-- CMAKE_SYSTEM_NAME: Linux
-- CMAKE_SYSTEM_PROCESSOR: x86_64
-- CMAKE_SYSTEM_NAME: Linux
-- CMAKE_SYSTEM_PROCESSOR: x86_64
-- Found local downloaded onnxruntime: /home/sunty/workspace/k2_env/sherpa-onnx/onnxruntime-linux-x64-gpu-1.15.1.tgz
-- Downloading onnxruntime from /home/sunty/workspace/k2_env/sherpa-onnx/onnxruntime-linux-x64-gpu-1.15.1.tgz
-- onnxruntime is downloaded to /home/sunty/workspace/k2_env/sherpa-onnx/build/_deps/onnxruntime-src
-- location_onnxruntime: /home/sunty/workspace/k2_env/sherpa-onnx/build/_deps/onnxruntime-src/lib/libonnxruntime.so
-- onnxruntime lib files: /home/sunty/workspace/k2_env/sherpa-onnx/build/_deps/onnxruntime-src/lib/libonnxruntime.so;/home/sunty/workspace/k2_env/sherpa-onnx/build/_deps/onnxruntime-src/lib/libonnxruntime.so.1.15.1;/home/sunty/workspace/k2_env/sherpa-onnx/build/_deps/onnxruntime-src/lib/libonnxruntime_providers_cuda.so;/home/sunty/workspace/k2_env/sherpa-onnx/build/_deps/onnxruntime-src/lib/libonnxruntime_providers_shared.so;/home/sunty/workspace/k2_env/sherpa-onnx/build/_deps/onnxruntime-src/lib/libonnxruntime_providers_tensorrt.so
-- Found local downloaded portaudio: /home/sunty/workspace/k2_env/sherpa-onnx/pa_stable_v190700_20210406.tgz
-- Downloading portaudio from /home/sunty/workspace/k2_env/sherpa-onnx/pa_stable_v190700_20210406.tgz
-- portaudio is downloaded to /home/sunty/workspace/k2_env/sherpa-onnx/build/_deps/portaudio-src
-- portaudio's binary dir is /home/sunty/workspace/k2_env/sherpa-onnx/build/_deps/portaudio-build
CMake Deprecation Warning at build/_deps/portaudio-src/CMakeLists.txt:7 (CMAKE_MINIMUM_REQUIRED):
  Compatibility with CMake < 2.8.12 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


CMake Warning (dev) at /usr/local/cmake/share/cmake-3.25/Modules/FindPackageHandleStandardArgs.cmake:438 (message):
  The package name passed to `find_package_handle_standard_args` (jack) does
  not match the name of the calling package (Jack).  This can lead to
  problems in calling code that expects `find_package` result variables
  (e.g., `_FOUND`) to follow a certain pattern.
Call Stack (most recent call first):
  build/_deps/portaudio-src/cmake_support/FindJack.cmake:34 (find_package_handle_standard_args)
  build/_deps/portaudio-src/CMakeLists.txt:277 (FIND_PACKAGE)
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Could NOT find jack (missing: JACK_LIBRARY JACK_INCLUDE_DIR) 
-- Could NOT find ALSA (missing: ALSA_LIBRARY ALSA_INCLUDE_DIR) 
-- Found local downloaded websocketpp: /home/sunty/workspace/k2_env/sherpa-onnx/websocketpp-b9aeec6eaf3d5610503439b4fae3581d9aff08e8.zip
-- Downloading websocketpp from /home/sunty/workspace/k2_env/sherpa-onnx/websocketpp-b9aeec6eaf3d5610503439b4fae3581d9aff08e8.zip
-- websocketpp is downloaded to /home/sunty/workspace/k2_env/sherpa-onnx/build/_deps/websocketpp-src
-- Found local downloaded asio: /home/sunty/workspace/k2_env/sherpa-onnx/asio-asio-1-24-0.tar.gz
-- Downloading asio /home/sunty/workspace/k2_env/sherpa-onnx/asio-asio-1-24-0.tar.gz
-- asio is downloaded to /home/sunty/workspace/k2_env/sherpa-onnx/build/_deps/asio-src
-- Found local downloaded json: /home/sunty/workspace/k2_env/sherpa-onnx/json-3.11.2.tar.gz
-- Downloading json from /home/sunty/workspace/k2_env/sherpa-onnx/json-3.11.2.tar.gz
-- json is downloaded to /home/sunty/workspace/k2_env/sherpa-onnx/build/_deps/json-src
-- Downloading cargs https://github.com/likle/cargs/archive/refs/tags/v1.0.3.tar.gz
-- cargs is downloaded to /home/sunty/workspace/k2_env/sherpa-onnx/build/_deps/cargs-src
-- CMAKE_CXX_FLAGS: 
-- Configuring done
-- Generating done
-- Build files have been written to: /home/sunty/workspace/k2_env/sherpa-onnx/build
```
and the output of make is:
```
make -j 4
[  0%] Building C object _deps/cargs-build/CMakeFiles/cargs.dir/src/cargs.c.o
[  1%] Building CXX object _deps/kaldi_native_fbank-build/kaldi-native-fbank/csrc/CMakeFiles/kaldi-native-fbank-core.dir/feature-fbank.cc.o
[  2%] Building CXX object _deps/kaldi_native_fbank-build/kaldi-native-fbank/csrc/CMakeFiles/kaldi-native-fbank-core.dir/feature-functions.cc.o
[  3%] Building C object _deps/portaudio-build/CMakeFiles/portaudio.dir/src/common/pa_allocation.c.o
[  4%] Building C object _deps/portaudio-build/CMakeFiles/portaudio.dir/src/common/pa_converters.c.o
[  5%] Linking C shared library ../../lib/libcargs.so
[  6%] Building CXX object _deps/kaldi_native_fbank-build/kaldi-native-fbank/csrc/CMakeFiles/kaldi-native-fbank-core.dir/feature-window.cc.o
[  6%] Built target cargs
[  7%] Building C object _deps/portaudio-build/CMakeFiles/portaudio.dir/src/common/pa_cpuload.c.o
[  8%] Building C object _deps/portaudio-build/CMakeFiles/portaudio.dir/src/common/pa_debugprint.c.o
[  9%] Building C object _deps/kaldi_native_fbank-build/kaldi-native-fbank/csrc/CMakeFiles/kaldi-native-fbank-core.dir/fftsg.c.o
[ 10%] Building C object _deps/portaudio-build/CMakeFiles/portaudio.dir/src/common/pa_dither.c.o
[ 11%] Building C object _deps/portaudio-build/CMakeFiles/portaudio.dir/src/common/pa_front.c.o
[ 12%] Building C object _deps/portaudio-build/CMakeFiles/portaudio.dir/src/common/pa_process.c.o
[ 13%] Building C object _deps/portaudio-build/CMakeFiles/portaudio.dir/src/common/pa_ringbuffer.c.o
[ 14%] Building C object _deps/portaudio-build/CMakeFiles/portaudio.dir/src/common/pa_stream.c.o
[ 15%] Building C object _deps/portaudio-build/CMakeFiles/portaudio.dir/src/common/pa_trace.c.o
[ 16%] Building C object _deps/portaudio-build/CMakeFiles/portaudio.dir/src/hostapi/skeleton/pa_hostapi_skeleton.c.o
[ 17%] Building C object _deps/portaudio-build/CMakeFiles/portaudio.dir/src/os/unix/pa_unix_hostapis.c.o
[ 18%] Building C object _deps/portaudio-build/CMakeFiles/portaudio.dir/src/os/unix/pa_unix_util.c.o
[ 20%] Building CXX object _deps/kaldi_native_fbank-build/kaldi-native-fbank/csrc/CMakeFiles/kaldi-native-fbank-core.dir/online-feature.cc.o
[ 20%] Building CXX object _deps/kaldi_native_fbank-build/kaldi-native-fbank/csrc/CMakeFiles/kaldi-native-fbank-core.dir/mel-computations.cc.o
[ 21%] Building CXX object _deps/kaldi_native_fbank-build/kaldi-native-fbank/csrc/CMakeFiles/kaldi-native-fbank-core.dir/rfft.cc.o
[ 21%] Linking C shared library ../../lib/libsherpa-onnx-portaudio.so
[ 21%] Built target portaudio
[ 22%] Building CXX object _deps/kaldi_native_fbank-build/kaldi-native-fbank/csrc/CMakeFiles/kaldi-native-fbank-core.dir/whisper-feature.cc.o
[ 23%] Linking CXX shared library ../../../../lib/libkaldi-native-fbank-core.so
[ 23%] Built target kaldi-native-fbank-core
[ 24%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/cat.cc.o
[ 25%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/base64-decode.cc.o
[ 26%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/context-graph.cc.o
[ 27%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/endpoint.cc.o
[ 28%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/features.cc.o
[ 29%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/file-utils.cc.o
[ 30%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/hypothesis.cc.o
[ 31%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/offline-ctc-greedy-search-decoder.cc.o
[ 32%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/offline-ctc-model.cc.o
[ 33%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/offline-lm-config.cc.o
[ 34%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/offline-lm.cc.o
[ 35%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/offline-model-config.cc.o
[ 36%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/offline-nemo-enc-dec-ctc-model-config.cc.o
[ 37%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/offline-nemo-enc-dec-ctc-model.cc.o
[ 38%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/offline-paraformer-greedy-search-decoder.cc.o
[ 39%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/offline-paraformer-model-config.cc.o
[ 40%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/offline-paraformer-model.cc.o
[ 41%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/offline-recognizer-impl.cc.o
[ 42%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/offline-recognizer.cc.o
[ 43%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/offline-rnn-lm.cc.o
[ 44%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/offline-stream.cc.o
[ 44%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/offline-transducer-greedy-search-decoder.cc.o
[ 45%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/offline-transducer-model-config.cc.o
[ 46%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/offline-transducer-model.cc.o
[ 47%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/offline-transducer-modified-beam-search-decoder.cc.o
[ 48%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/offline-whisper-greedy-search-decoder.cc.o
[ 49%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/offline-whisper-model-config.cc.o
[ 50%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/offline-whisper-model.cc.o
[ 51%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/online-conformer-transducer-model.cc.o
[ 52%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/online-lm-config.cc.o
[ 53%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/online-lm.cc.o
[ 54%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/online-lstm-transducer-model.cc.o
[ 55%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/online-recognizer.cc.o
[ 56%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/online-rnn-lm.cc.o
[ 57%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/online-stream.cc.o
[ 58%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/online-transducer-decoder.cc.o
[ 59%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/online-transducer-greedy-search-decoder.cc.o
[ 60%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/online-transducer-model-config.cc.o
[ 61%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/online-transducer-model.cc.o
[ 62%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/online-transducer-modified-beam-search-decoder.cc.o
[ 63%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/online-zipformer-transducer-model.cc.o
[ 64%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/online-zipformer2-transducer-model.cc.o
[ 65%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/onnx-utils.cc.o
[ 66%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/packed-sequence.cc.o
[ 67%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/pad-sequence.cc.o
[ 68%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/parse-options.cc.o
[ 69%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/provider.cc.o
[ 69%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/resample.cc.o
[ 70%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/session.cc.o
[ 71%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/slice.cc.o
[ 72%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/stack.cc.o
[ 73%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/symbol-table.cc.o
In file included from /opt/rh/devtoolset-8/root/usr/include/c++/8/backward/strstream:50,
                 from /home/sunty/workspace/k2_env/sherpa-onnx/sherpa-onnx/csrc/symbol-table.cc:10:
/opt/rh/devtoolset-8/root/usr/include/c++/8/backward/backward_warning.h:32:2: warning: #warning This file includes at least one deprecated or antiquated header which may be removed without further notice at a future date. Please use a non-deprecated interface with equivalent functionality instead. For a listing of replacement headers and interfaces, consult the file backward_warning.h. To disable this warning use -Wno-deprecated. [-Wcpp]
 #warning \
  ^~~~~~~
[ 74%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/text-utils.cc.o
[ 75%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/transpose.cc.o
[ 76%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/unbind.cc.o
[ 77%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-core.dir/wave-reader.cc.o
[ 78%] Linking CXX shared library ../../lib/libsherpa-onnx-core.so
[ 78%] Built target sherpa-onnx-core
[ 80%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-offline.dir/sherpa-onnx-offline.cc.o
[ 80%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-microphone.dir/sherpa-onnx-microphone.cc.o
[ 81%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx.dir/sherpa-onnx.cc.o
[ 82%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-microphone-offline.dir/sherpa-onnx-microphone-offline.cc.o
[ 83%] Linking CXX executable ../../bin/sherpa-onnx-offline
[ 84%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-microphone-offline.dir/microphone.cc.o
../../_deps/onnxruntime-src/lib/libonnxruntime_providers_cuda.so: undefined reference to `Provider_GetHost'
collect2: error: ld returned 1 exit status
make[2]: *** [bin/sherpa-onnx-offline] Error 1
make[1]: *** [sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-offline.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 85%] Building CXX object sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-microphone.dir/microphone.cc.o
[ 86%] Linking CXX executable ../../bin/sherpa-onnx-microphone-offline
../../_deps/onnxruntime-src/lib/libonnxruntime_providers_cuda.so: undefined reference to `Provider_GetHost'
collect2: error: ld returned 1 exit status
make[2]: *** [bin/sherpa-onnx-microphone-offline] Error 1
make[1]: *** [sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-microphone-offline.dir/all] Error 2
[ 87%] Linking CXX executable ../../bin/sherpa-onnx-microphone
[ 88%] Linking CXX executable ../../bin/sherpa-onnx
../../_deps/onnxruntime-src/lib/libonnxruntime_providers_cuda.so: undefined reference to `Provider_GetHost'
collect2: error: ld returned 1 exit status
make[2]: *** [bin/sherpa-onnx-microphone] Error 1
make[1]: *** [sherpa-onnx/csrc/CMakeFiles/sherpa-onnx-microphone.dir/all] Error 2
../../_deps/onnxruntime-src/lib/libonnxruntime_providers_cuda.so: undefined reference to `Provider_GetHost'
collect2: error: ld returned 1 exit status
make[2]: *** [bin/sherpa-onnx] Error 1
make[1]: *** [sherpa-onnx/csrc/CMakeFiles/sherpa-onnx.dir/all] Error 2
make: *** [all] Error 2
```
I checked the missing symbol
```
(sherpa_hotword) [sunty@3090 lib]$ pwd
/home/sunty/workspace/k2_env/sherpa-onnx/build/_deps/onnxruntime-src/lib
(sherpa_hotword) [sunty@3090 lib]$ nm --dynamic ./libonnxruntime_providers_shared.so
                 w __cxa_finalize
                 w __gmon_start__
                 w _ITM_deregisterTMCloneTable
                 w _ITM_registerTMCloneTable
0000000000001110 T Provider_GetHost
0000000000001120 T Provider_SetHost
0000000000000000 A VERS_1.0
```

--- a/.github/workflows/linux-gpu.yaml
+++ b/.github/workflows/linux-gpu.yaml
@@ -1,0 +1,151 @@
+name: linux-gpu
+
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - '*'
+    paths:
+      - '.github/workflows/linux-gpu.yaml'
+      - '.github/scripts/test-online-transducer.sh'
+      - '.github/scripts/test-offline-transducer.sh'
+      - '.github/scripts/test-offline-ctc.sh'
+      - 'CMakeLists.txt'
+      - 'cmake/**'
+      - 'sherpa-onnx/csrc/*'
+      - 'sherpa-onnx/c-api/*'
+      - 'c-api-examples/**'
+  pull_request:
+    branches:
+      - master
+    paths:
+      - '.github/workflows/linux-gpu.yaml'
+      - '.github/scripts/test-online-transducer.sh'
+      - '.github/scripts/test-offline-transducer.sh'
+      - '.github/scripts/test-offline-ctc.sh'
+      - 'CMakeLists.txt'
+      - 'cmake/**'
+      - 'sherpa-onnx/csrc/*'
+      - 'sherpa-onnx/c-api/*'
+  release:
+    types:
+      - published
+
+  workflow_dispatch:
+    inputs:
+      release:
+        description: "Whether to release"
+        type: boolean
+
+env:
+  RELEASE:
+    |- # Release if there is a release tag name or a release flag in workflow_dispatch
+    ${{ github.event.release.tag_name != '' || github.event.inputs.release == 'true' }}
+
+concurrency:
+  group: linux-gpu-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  linux_gpu:
+    runs-on: ${{ matrix.os }}
+    name: ${{ matrix.build_type }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest]
+        build_type: [Release, Debug]
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Configure CMake
+        shell: bash
+        run: |
+          mkdir build
+          cd build
+          cmake -D CMAKE_BUILD_TYPE=${{ matrix.build_type }} -DCMAKE_INSTALL_PREFIX=./install -DBUILD_SHARED_LIBS=ON -DSHERPA_ONNX_ENABLE_GPU=ON ..
+
+      - name: Build sherpa-onnx for ubuntu
+        shell: bash
+        run: |
+          cd build
+          make -j2
+          make install
+
+          ls -lh lib
+          ls -lh bin
+
+      - name: Display dependencies of sherpa-onnx for linux
+        shell: bash
+        run: |
+          file build/bin/sherpa-onnx
+          readelf -d build/bin/sherpa-onnx
+
+      - name: Test offline Whisper
+        shell: bash
+        run: |
+          export PATH=$PWD/build/bin:$PATH
+          export EXE=sherpa-onnx-offline
+
+          .github/scripts/test-offline-whisper.sh
+
+      - name: Test offline CTC
+        shell: bash
+        run: |
+          export PATH=$PWD/build/bin:$PATH
+          export EXE=sherpa-onnx-offline
+
+          .github/scripts/test-offline-ctc.sh
+
+      - name: Test offline transducer
+        shell: bash
+        run: |
+          export PATH=$PWD/build/bin:$PATH
+          export EXE=sherpa-onnx-offline
+
+          .github/scripts/test-offline-transducer.sh
+
+      - name: Test online transducer
+        shell: bash
+        run: |
+          export PATH=$PWD/build/bin:$PATH
+          export EXE=sherpa-onnx
+
+          .github/scripts/test-online-transducer.sh
+
+      - name: Test online transducer (C API)
+        shell: bash
+        run: |
+          export PATH=$PWD/build/bin:$PATH
+          export EXE=decode-file-c-api
+
+          .github/scripts/test-online-transducer.sh
+
+      - name: Copy files
+        if: env.RELEASE == 'true'
+        shell: bash
+        run: |
+          SHERPA_ONNX_VERSION=v$(grep "SHERPA_ONNX_VERSION" ./CMakeLists.txt  | cut -d " " -f 2  | cut -d '"' -f 2)
+
+          dst=sherpa-onnx-${SHERPA_ONNX_VERSION}-linux-x64-gpu
+          mkdir $dst
+
+          cp -a build/install/bin $dst/
+          cp -a build/install/lib $dst/
+          cp -a build/install/include $dst/
+
+          tree $dst
+
+          tar cjvf ${dst}.tar.bz2 $dst
+
+      - name: Release pre-compiled binaries and libs for linux x64
+        if: env.RELEASE == 'true' && matrix.build_type == 'Release'
+        uses: svenstaro/upload-release-action@v2
+        with:
+          file_glob: true
+          overwrite: true
+          file: sherpa-onnx-*linux-x64-gpu.tar.bz2

--- a/cmake/onnxruntime-linux-x86_64-gpu.cmake
+++ b/cmake/onnxruntime-linux-x86_64-gpu.cmake
@@ -82,6 +82,19 @@ add_library(onnxruntime_providers_cuda SHARED IMPORTED)
 set_target_properties(onnxruntime_providers_cuda PROPERTIES
   IMPORTED_LOCATION ${location_onnxruntime_cuda_lib}
 )
+message(STATUS "location_onnxruntime_cuda_lib: ${location_onnxruntime_cuda_lib}")
+
+# for libonnxruntime_providers_shared.so
+find_library(location_onnxruntime_providers_shared_lib onnxruntime_providers_shared
+  PATHS
+  "${onnxruntime_SOURCE_DIR}/lib"
+  NO_CMAKE_SYSTEM_PATH
+)
+add_library(onnxruntime_providers_shared SHARED IMPORTED)
+set_target_properties(onnxruntime_providers_shared PROPERTIES
+  IMPORTED_LOCATION ${location_onnxruntime_providers_shared_lib}
+)
+message(STATUS "location_onnxruntime_providers_shared_lib: ${location_onnxruntime_providers_shared_lib}")
 
 file(GLOB onnxruntime_lib_files "${onnxruntime_SOURCE_DIR}/lib/libonnxruntime*")
 message(STATUS "onnxruntime lib files: ${onnxruntime_lib_files}")

--- a/cmake/onnxruntime-win-x64-gpu.cmake
+++ b/cmake/onnxruntime-win-x64-gpu.cmake
@@ -82,9 +82,18 @@ file(COPY ${onnxruntime_SOURCE_DIR}/lib/onnxruntime.dll
     ${CMAKE_BINARY_DIR}/bin/${CMAKE_BUILD_TYPE}
 )
 
+# for onnxruntime_providers_cuda.dll
+
+find_library(location_onnxruntime_providers_cuda_lib onnxruntime_providers_cuda
+  PATHS
+  "${onnxruntime_SOURCE_DIR}/lib"
+  NO_CMAKE_SYSTEM_PATH
+)
+message(STATUS "location_onnxruntime_providers_cuda_lib: ${location_onnxruntime_providers_cuda_lib}")
+
 add_library(onnxruntime_providers_cuda SHARED IMPORTED)
 set_target_properties(onnxruntime_providers_cuda PROPERTIES
-  IMPORTED_LOCATION ${location_onnxruntime}
+  IMPORTED_LOCATION ${location_onnxruntime_providers_cuda_lib}
   INTERFACE_INCLUDE_DIRECTORIES "${onnxruntime_SOURCE_DIR}/include"
 )
 
@@ -93,7 +102,28 @@ set_property(TARGET onnxruntime_providers_cuda
     IMPORTED_IMPLIB "${onnxruntime_SOURCE_DIR}/lib/onnxruntime_providers_cuda.lib"
 )
 
-file(COPY ${onnxruntime_SOURCE_DIR}/lib/onnxruntime_providers_cuda.dll
+# for onnxruntime_providers_shared.dll
+
+find_library(location_onnxruntime_providers_shared_lib onnxruntime_providers_shared
+  PATHS
+  "${onnxruntime_SOURCE_DIR}/lib"
+  NO_CMAKE_SYSTEM_PATH
+)
+message(STATUS "location_onnxruntime_providers_shared_lib: ${location_onnxruntime_providers_shared_lib}")
+add_library(onnxruntime_providers_shared SHARED IMPORTED)
+set_target_properties(onnxruntime_providers_shared PROPERTIES
+  IMPORTED_LOCATION ${location_onnxruntime_providers_shared_lib}
+  INTERFACE_INCLUDE_DIRECTORIES "${onnxruntime_SOURCE_DIR}/include"
+)
+set_property(TARGET onnxruntime_providers_shared
+  PROPERTY
+    IMPORTED_IMPLIB "${onnxruntime_SOURCE_DIR}/lib/onnxruntime_providers_shared.lib"
+)
+
+file(
+  COPY
+    ${onnxruntime_SOURCE_DIR}/lib/onnxruntime_providers_cuda.dll
+    ${onnxruntime_SOURCE_DIR}/lib/onnxruntime_providers_shared.dll
   DESTINATION
     ${CMAKE_BINARY_DIR}/bin/${CMAKE_BUILD_TYPE}
 )

--- a/sherpa-onnx/csrc/CMakeLists.txt
+++ b/sherpa-onnx/csrc/CMakeLists.txt
@@ -90,6 +90,7 @@ endif()
 if(SHERPA_ONNX_ENABLE_GPU)
   target_link_libraries(sherpa-onnx-core
     onnxruntime_providers_cuda
+    onnxruntime_providers_shared
   )
 endif()
 

